nrf_security/Kconfig

```diff
@@ -0,0 +1,477 @@
+menu "Nordic Security"
+
+config NORDIC_SECURITY_BACKEND
+	bool
+	prompt "Use Nordic provided security backend"
+	depends on SOC_FAMILY_NRF
+	select NRFXLIB_CRYPTO
+	help
+	  Help message
+
+if NORDIC_SECURITY_BACKEND
+
+comment "Backend Selection"
+
+config CC310_BACKEND
+	bool "Use cc310 backend"
+	default y
+	depends on SOC_NRF52840
+	select NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0 if VANILLA_BACKEND
+	help
+	  Enable cc310 backend
+
+config VANILLA_BACKEND
+	bool "Use mbedtls backend"
+	default y if !(SOC_NRF52840 || SOC_NRF9160)
+	help
+	  Enable mbedtls vanilla backend
+
+# Group 0 is used to indicate when AES and ECC should use choice menu
+config NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
```

Comment:
From what I've seen in the code base, this config is used for a lot more than the comment says. I suggest giving it a more accurate name, and a slightly longer comment :). (The comment could be `help` text BTW)

Reply:
Good point.

The comment should be updated.
For future updates to security features I prefer to keep name independent of actual selected backends, as it will allow for additional backends to be part of the choice group.
However, maybe the following name would be slightly better:
`NRF_CRYPTO_MULTIPLE_BACKEND_SELECTION_0`

If name starts to contain backend knowledge, like `NRF_CRYPTO_CC310_AND_VANILLA`, then the name must also change when adding more backends.

Reply:
I still don't really know what it means, but I'm guessing it means both backends are enabled? And if you had 3 backends, you would have a config for each combination of backends?

How about `NRF_CRYPTO_BACKEND_COMBINATION_0`

or `NRF_CRYPTO_BACKEND_COMBINATION` and it has a number value like 

    NRF_CRYPTO_BACKEND_COMBINATION = BACKEND_VANILLA + (BACKEND_CC310 << 1)

Reply:
`NRF_CRYPTO_BACKEND_COMBINATION_0` is also fine.

`NRF_CRYPTO_BACKEND_COMBINATION = BACKEND_VANILLA + (BACKEND_CC310 << 1)` 
will not achieve what is already prepared for, as it will require more logic later in config selection as not all backends support the same, thus it still requires the logic we try to avoid with this `group` symbol.

Reply:
> And if you had 3 backends, you would have a config for each combination of backends?

Not necessarily, the group is more a combination of supported features and the backends selected that support the particular feature set.
In this case the `NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0` it indicates that multiple backends has been selected which all support AES and ECC.